.github/workflows/tests.yml

```diff
@@ -0,0 +1,105 @@
+name: 'Tests'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.ref }}
+  cancel-in-progress: true
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+permissions:
+  contents: read
+
+jobs:
+  pytest:
+    name: '${{ matrix.os }} / ${{ matrix.python }} / ${{ matrix.distrib }} / ${{ matrix.sphinx_version }}'
+    timeout-minutes: 30
+    continue-on-error: true
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -el {0}
+    env:
+      PYTHON_VERSION: '${{ matrix.python }}'
+      SPHINX_VERSION: '${{ matrix.sphinx_version }}'
+      DISTRIB: '${{ matrix.distrib }}'
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            python: '3.12'
+            sphinx_version: dev
+            distrib: mamba
+            locale: C
+          - os: ubuntu-latest
+            python: '3.11'
+            sphinx_version: '6'
```

Comment:
Just a question, why do want to test this sphinx version?

Reply:
Really just to try to hit all supported major Sphinx versions but I'll make it clearer